File: Python/M01_ProgrammingBasics/L06_NestedLoops/Exercises/Solutions/P04_TrainTheTrainers.py
people = int(input())
name_doc = input()
grade_sum = 0
average_grade = 0
total_grade = 0
numbers = 0

while name_doc != "Finish":
    for x in range(people):
        grade = float(input())
        grade_sum += grade
        average_grade = grade_sum / people
    print(f"{name_doc} - {average_grade:.2f}.")
    name_doc = input()
    grade_sum = 0
    total_grade += average_grade
    numbers += 1

print(f"Student's final assessment is {total_grade / numbers:.2f}.")

File: C++/M01_ProgrammingBasics/L06_NestedLoops/Exercises/Solutions/P04_TrainTheTrainers.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int countJury;
    cin >> countJury;
    cin.ignore();

    cout.setf(ios::fixed);
    cout.precision(2);

    string input;
    getline(cin, input);

    double sumAllGrades = 0;
    int countAllGrades = 0;
    while (input != "Finish") {
        double sumGradesPerPresentation = 0; 
        for (int jury = 1; jury <= countJury; jury++) {
            double grade;
            cin >> grade;

            sumGradesPerPresentation += grade; 

            sumAllGrades += grade; 
            countAllGrades++; 
        }
        double averageGradePerPresentation = sumGradesPerPresentation / countJury;
        cout << input << " - " << averageGradePerPresentation << "." << endl;

        cin.ignore();
        getline(cin, input);
    }

    double averageAll = sumAllGrades / countAllGrades;
    cout << "Student's final assessment is " << averageAll << "." << endl;

    return 0;
}